languages/fr.lua

```diff
@@ -172,7 +172,7 @@ function SILE.nodeMakers.fr:mustRemove (i, items)
   -- Clear "manual" spaces we do not want, so that later we only have to
   -- insert the relevant kerns.
   local curr = items[i].text
-  if self:isSpace(curr) then
+  if self:isSpace(curr) or self:isNonBreakingSpace(curr) then
```

Comment:
So this is the place where we remove some spaces from the input, to later add correct French punctuation spacing.
I wonder, perhaps we should probably also check U+202F (narrow no-break space) and U+2009 (thin space) here. Some (French ?) people think they are doing good typography by using them in their input documents, but we'd likely prefer all of these be _also_ removed and replaced via our French punctuation support logic?

Reply:
On the other hand, the rationale is not quite the same. I'd say "let U+202F and U+2009 live for now as they are, and wait until another Frenchy than I has a strong opinion on these" ;-)

Reply:
I have enough to do keeping up with work with, much less ones I don't. I'm fine with leaving this up to the Frenchies. That being said an issue should probably be opened with these thought since this thread will be pretty much burred soon.

Reply:
> That being said an issue should probably be opened with these thought since this thread will be pretty much burred soon.

Opened as #1945